server/src/main/java/de/uftos/services/TimetableService.java

```diff
@@ -276,28 +278,74 @@ private List<GradeProblemDto> getGrades() {
   private List<LessonProblemDto> getLessons(Timetable timetable) {
     //initializing new Lessons
     List<Curriculum> curriculums = curriculumRepository.findAll();
-    for (StudentGroup studentGroup : studentGroupRepository.findAll()) {
+    if (curriculums.isEmpty()) {
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+          "There must be at least one curriculum to create a timetable");
+    }
+
+    List<StudentGroup> studentGroups = studentGroupRepository.findAll();
+    if (studentGroups.isEmpty()) {
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+          "There must be at least one student group to create a timetable");
+    }
+    for (StudentGroup studentGroup : studentGroups) {
       for (Curriculum curriculum : curriculums) {
         if (curriculum.getGrade() != studentGroup.getGrades().getFirst()) {
           continue;
         }
-        for (LessonsCount lessonsCount : curriculum.getLessonsCounts()) {
+
+        List<LessonsCount> lessonsCounts = curriculum.getLessonsCounts();
+        // TODO is this needed?
+        if (lessonsCounts.isEmpty()) {
+          throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+              "There must be at least one lesson set in a curriculum to create a timetable");
```

Comment:
This should probably say something like "There must be at least on existing subject!"
Although this will not prevent, that all lesson counts are 0